shell.nix

```diff
@@ -27,18 +27,38 @@ let
   '';
 
   kapply = pkgs.writeShellScriptBin "kapply" ''
+    #!/bin/bash
     set -e
     set -o pipefail
     target="$1"
     shift
     manifests="$(mktemp)"
     kustomize build "$target" > "$manifests"
-    kubectl diff -f "$manifests" "$@"
+    kubectl diff -f "$manifests" "$@" || echo $?
```

Comment:
When the context is wrong, then it's likely that the diff already fails. E.g. the namespace might not exist. It's better to specify the context on the command line or infer it automatically.

Reply:
That is why we just print wrong exit codes instead of crashing.
But I will check how much work it is to automatically infer it.